### ISSUE TITLE
Some suggestions and questions

### ISSUE THREAD

**Issue by sailxjx** [Unitech/pm2-deploy#27]
1. Is "deploy with git clone" a good idea? Or "should I keep a .git directory on the server"? I doubt it. I prefer "clone the repos locally and deploy the source code with some tool like `rsync`", I can help do some changes to the code if you like.
2. The "current" link always refer to to the "source" directory, this make the link useless. And when you need to rollback, you should run the deploy process (and the endless "pre-deploy" and "post-deploy") again, even though, you may not revert to the exactly same dependencies of the right version. If you try to keep all versions on the disk, you rollback will be extremely easy:

```
Project
- current -> source/20141118120121
- shared
- source
  - 20141118120121
  - 20141117112012
  - ...
```
1. Embed "ecosystem.json" in each deploy command is annoying me. For the reason of "convention over configuration", just use the "ecosystem.json" file in the current directory (No one likes to use the `npm` command as "npm install package.json"). If you worry about the conflict of file names, choose another name (such as "pm2eco.json").
2. I am confused by the structure of configuration file. The deploy command only use the <environment> variable, but not name of application. Most of time I will only deploy one project to the server, for example in the default ecosystem.json

``` json
{
  "apps" : [{
    "name"      : "API",
    "script"    : "app.js",
    "env": {
      "COMMON_VARIABLE": "true"
    },
    "env_production" : {
      "NODE_ENV": "production"
    }
  },{
    "name"      : "WEB",
    "script"    : "web.js"
  }],
  "deploy" : {
    "production" : {
      "user" : "node",
      "host" : "212.83.163.1",
      "ref"  : "origin/master",
      "repo" : "git@github.com:repo.git",
      "path" : "/var/www/production",
      "post-deploy" : "pm2 startOrRestart ecosystem.json --env production"
    },
    "dev" : {
      "user" : "node",
      "host" : "212.83.163.1",
      "ref"  : "origin/master",
      "repo" : "git@github.com:repo.git",
      "path" : "/var/www/development",
      "post-deploy" : "pm2 startOrRestart ecosystem.json --env dev"
    }
  }
}
```

I want deploy the API and WEB separately, but `pm2 deploy` don't give this try. And if I want to set "instances: 2" in "env_test" and "instance: 8" in "env_production", the format of configuration file may not support this way.

So, if I said something wrong, please point it out.

**Comment by Vanthiyadevan** [comment]
I agree with @sailxjx  prefer not to have a .git repository on the server.

**Comment by Vanthiyadevan** [comment]
#168

**Comment by Vanthiyadevan** [comment]
#168

**Comment by sailxjx** [comment]
@devansvd I'm surprised to find this issue still opened, as an alternative you can give a try to my project https://github.com/teambition/sneaky.

I don't have much time to maintain this project, fortunately this project is very simple and the source code is really short.

Though my team have changed most of our projects to `docker/kubernetes` stack, this tool is still in use in some projects.

**Comment by devansvd** [comment]
@sailxjx thanks will check them out.
